code/hsec-sync/src/Security/Advisories/Sync/Snapshot.hs

```diff
@@ -0,0 +1,181 @@
+{-# LANGUAGE DeriveAnyClass #-}
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingStrategies #-}
+{-# LANGUAGE LambdaCase #-}
+
+-- |
+--
+-- Helpers for deriving advisory metadata from a Snapshot s.
+module Security.Advisories.Sync.Snapshot
+  ( SnapshotDirectoryInfo (..),
+    SnapshotError (..),
+    explainSnapshotError,
+    Snapshot (..),
+    SnapshotRepositoryEnsuredStatus (..),
+    ensureSnapshot,
+    getDirectorySnapshotInfo,
+    overwriteSnapshot,
+    SnapshotRepositoryStatus (..),
+    snapshotRepositoryStatus,
+  )
+where
+
+import qualified Codec.Archive.Tar as Tar
+import qualified Codec.Compression.GZip as GZip
+import Control.Exception (IOException, try)
+import Control.Lens
+import Control.Monad.Extra (unlessM, whenM)
```

Comment:
these are `when` and `unless`, you don't need the `Monad` instance, `Applicative` is enough.

Reply:
I did it to avoid excessive bindings, such as:

```haskell
    unlessM (liftIO $ D.doesFileExist metadataPath) $
      throwE SnapshotDirectoryMissingE
```

Reply:
ah! sure; I normally use the `>>= flip` pattern on these.